components/header.js

```diff
@@ -1,16 +1,199 @@
-import React from 'react';
+'use client';
+import React, { useState } from 'react';
 import pageStyles from '../app/page.module.css';
+import { icons } from '@/components/utilities/card';
+import NavHeader from './utilities/navHeader';
 
 const Header = () => {
+  const [headerRules, setHeaderRules] = useState({ cardRank: false, suitRank: false, combinations: false, smashes: false, gameplay: false });
+  const [toggleRules, setToggleRules] = useState(false);
+
+  const handleClick = (type) => {
+    setHeaderRules({
+      ...headerRules,
+      [type]: !headerRules[type],
+    });
+  }
+
   return (
     <header className={pageStyles.header}>
       <h1>Thirteen</h1>
       <nav>
         <ul>
-          <li><button>Rules</button></li>
+          <li><button onClick={() => setToggleRules(true)}>Rules</button></li>
           <li><button>Dark Mode</button></li>
         </ul>
       </nav>
+      <div className={toggleRules ? pageStyles.rules : pageStyles.hide}>
+        <div className={pageStyles.rulesHeader}>
```

Comment:
i would consider making the header of the modal sticky to the top. in the scenario where a user may open all the accordions to read everything, they would be forced to scroll all the way up to close it. if we make this sticky, they can just close it wherever they are on the screen